_posts/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that.md

```diff
@@ -0,0 +1,325 @@
+    const button = wrapper.find('button').at(1);
+
+    // when
+    button.simulate('click');
+
+    // then
+    expect(wrapper.find('.counter').text()).toBe('99');
+  });
+});
+
+
+```
+### What need to be change:
+- remove **mount** and use **render** instead
+- remove **wrapper** variable and use **screen** to get elements
+- replace **simulate(’click’)** with **userEvent**
+
+### React testing library example: 
+```js
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { Test } from '../Test';
+
+describe('<Test />', () => {
+  it('should render default header', () => {
+    // when
+    render(<Test />);
+
+    // then
+    expect(screen.getByRole('heading', { name: 'Default header' })).toBeInTheDocument();
+  });
+  it('should render header from props', () => {
+    // when
+    render(<Test header="My simple calculator" />);
+
+    // then
+    expect(screen.getByRole('heading', { name: 'My simple calculator' })).toBeInTheDocument();
+    expect(screen.queryByText('Default header')).not.toBeInTheDocument();
+  });
+  it('should render default defaultCounter', () => {
+    // when
+    render(<Test />);
+
+    // then
+    expect(screen.getByText('0')).toBeInTheDocument();
+  });
+  it('should render defaultCounter from props', () => {
+    // when
+    render(<Test defaultCounter={100} />);
+
+    // then
+    expect(screen.getByText('100')).toBeInTheDocument();
+  });
+  it('should increment conunter on click', () => {
+    // given
+    render(<Test defaultCounter={100} />);
+
+    const button = screen.getByRole('button', { name: 'Increment counter' });
+
+    // when
+    userEvent.click(button);
+
+    // then
+    expect(screen.getByText('101')).toBeInTheDocument();
+  });
+  it('should decrement conunter on click', () => {
+    // given
+    render(<Test defaultCounter={100} />);
+
+    const button = screen.getByRole('button', { name: 'Decrement counter' });
+
+    // when
+    userEvent.click(button);
+
+    // then
+    expect(screen.getByText('99')).toBeInTheDocument();
+  });
+});
+
+```
+## Enzyme and RTL in Allegro Ads
+
+My team is currently working on the development of [Allegro Ads](https://allegro.pl/ads), which is an advertising tool 
+for our ecommerce platform sellers. It is a whole panel for viewing statistics, displaying advertising campaigns. We 
+provide sponsored offers for our allegro site, facebook, google and recently even for on the partner websites. Our 
+sellers pay us for the opportunity to advertise their products, and at the same time we strive to ensure that these 
+ads accurately reach people potentially interested in buying, thereby generating profits for sellers.
+
+The main technology stack is standard and still pretty cool. The core of the project is written in React, and 
+significant parts have been rewritten to Typescript. From the beginning, for testing purposes, we use Jest. In 
```

Comment:
We don't use Jest from the beginning 😄  I rewrote it from Mocha/Chai/Sinon about a year ago

Reply:
Oh, good to know that!!!